File: android/src/main/java/io/flutter/push/FirebaseInstanceIdListenerService.java
package io.flutter.push;

import com.google.firebase.iid.FirebaseInstanceIdService;

/**
 * Created by ricardo on 12/12/16.
 */

public class FirebaseInstanceIdListenerService extends FirebaseInstanceIdService {
    @Override
    public void onTokenRefresh() {
        PushPlugin.executeOnTokenRefreshCallback();
    }

}

File: android/src/main/kotlin/io/flutter/plugins/push/extra/FirebaseInstanceIdListenerService.kt
package io.flutter.plugins.push.extra

import android.util.Log
import com.google.firebase.messaging.FirebaseMessagingService

/**
 * Created by ricardo on 12/12/16.
 */

class FirebaseInstanceIdListenerService : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        Log.e("NEW_TOKEN", token);
        PushPlugin.executeOnTokenRefreshCallback()
    }
}